src/App.tsx

```diff
@@ -1,14 +1,46 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoadedTodos, setIsLoadedTodos] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [inputState, setInputState] = useState('');
+  const [toggleFilter, setToggleFilter] = useState<boolean | null>(null);
+
+  let filteredTodos = [...todos];
+
+  const fetchData = useCallback(async () => {
+    const data = await getTodos();
+
+    setTodos(data);
+    setIsLoadedTodos(true);
+  }, []);
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  if (inputState && inputState.length > 0) {
```

Comment:
just one evaluation is sufficient
since initial state (empty string `''`) is a false equivalent
```suggestion
  if (inputState) {
```